### PR TITLE
✨ `thermostat` card: follow same step logic as more-info

### PR DESCRIPTION
**Pull request by iantrich**
https://community.home-assistant.io/t/lovelace-thermostat-add-support-for-target-temp-step/76732

### CODE REVIEW

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 1]
```diff
@@ -204,13 +204,25 @@ export class HuiThermostatCard extends hassLocalizeLitMixin(LitElement)
       min: stateObj.attributes.min_temp,
       max: stateObj.attributes.max_temp,
       sliderType: _sliderType,
+      step: this._computeTemperatureStepSize(),
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: I think we should set this value before the jquery slider set up. and also take step out of the defaults in the top of the file

Comment (iantrich) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: I agree with removing the default, I missed that, but not sure what you mean by setting before the slider setup.

Comment (iantrich) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: Are you just saying to do a
```
const step = this._computeTemperatureStepSize();
```
to pass to the constructor?

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 2]
```diff
@@ -207,10 +207,22 @@ export class HuiThermostatCard extends hassLocalizeLitMixin(LitElement)
       change: (value) => this._setTemperature(value),
       drag: (value) => this._dragEvent(value),
       value: sliderValue,
+      step,
     });
     this._updateSetTemp(uiValue);
   }
 
+  private _computeTemperatureStepSize() {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: extract this as a function and put above this class definition.

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 3]
```diff
@@ -207,10 +207,22 @@ export class HuiThermostatCard extends hassLocalizeLitMixin(LitElement)
...
+    const stateObj = this.hass!.states[this._config!.entity];
+    if (stateObj.attributes.target_temp_step) {
+      return stateObj.attributes.target_temp_step;
+    }
+    if (this.hass!.config.unit_system.temperature.indexOf("F") !== -1) {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: ??

Just use `===` and a ternary statement

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 4]
```diff
@@ -55,6 +54,14 @@ function formatTemp(temps: string[]): string {
   return temps.filter(Boolean).join("-");
 }
 
+function computeTemperatureStepSize(hass: HomeAssistant, config: Config) {
+  const stateObj = hass.states[config.entity];
+  if (stateObj.attributes.target_temp_step) {
+    return stateObj.attributes.target_temp_step;
+  }
+  return hass.config.unit_system.temperature.indexOf("F") === 1 ? 1 : 0.5;
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: We shouldn't do indexOf, we should do `=== UNIT_F` (already defined in `const.ts`)

**Pull request merged**